sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -386,8 +386,8 @@ class Analyzer(override val catalogManager: CatalogManager)
             DatetimeSub(l, r, DateAddInterval(l, UnaryMinus(r, f), ansiEnabled = f))
           case (_, CalendarIntervalType | _: DayTimeIntervalType) =>
             Cast(DatetimeSub(l, r, TimeAdd(l, UnaryMinus(r, f))), l.dataType)
-          case (TimestampType, _) => SubtractTimestamps(l, r)
-          case (_, TimestampType) => SubtractTimestamps(l, r)
+          case (TimestampType | TimestampWithoutTZType, _) => SubtractTimestamps(l, r)
```

Comment:
Can this be `case (AnyTimestampType(), _)`?

Reply:
No. But we can simplify it with 
```
case _ if AnyTimestampType.unapply(l) || AnyTimestampType.unapply(r) =>
```